### DISCUSSION TITLE
Regarding running WWIII in parallel mode

### DISCUSSION THREAD

**Discussion by krishnanathira** [NOAA-EMC/WW3#1042] [Q&A] [answered]
Hi 
Can you guide me the right way to run WWIII model in MPI mode. I am running using qsub command.
Is it okay to run the grid.nml, strt.inp and prnc.nml directly and shel.nml alone in the qsub mode.

Or is it preferred to run using the run_test in the bin folder. 
`./bin/run_test -n 16 -p mpirun -s Ifremer1 -w output_IO ../model IO` 
this way?

Thanks
Athira

**Comment by aliabdolali** [comment]
The only parallel executables of ww3 are ww3_shel and ww3_multi. Even if you allocate more than one core to the rest, one would be utilized.
The run_test is designed to execute them in order, like pre_processing ones such as ww3_grid first, then ww3_prnc,... and eventually ww3_shel. The post processing ones will be executed once the simulations is over.

**Reply by krishnanathira** [reply]
Thanks for the quick response.
May I know whether the run_test will work for shel.nml as well? Or is it required to be in .inp format?

Thanks

**Comment by aliabdolali** [comment, marked as answer]
The run_tests will check all  available *.inp and *.nml files in the run directory and execute the corresponding executable in order.
However, in case both nml and inp file exists, it goes with inp, unless you use `-N` option to enforce it to utilize nml.
I'd suggest keeping one type per program (either *nml or *inp).
Also, if you execute `run_test -h` or `run_cmake_test -h`only, you can see all the options.

**Reply by krishnanathira** [reply]
Thank you very much.

Regards
Athira
